apps/aevm/src/aevm_eeevm_state.erl

```diff
@@ -167,7 +167,9 @@ init_vm(State, Code, Mem, Store) ->
                     {ok, CalldataHeap} = aeso_data:binary_to_heap({tuple, [typerep, Type]}, Calldata,
                                                                   aevm_eeevm_maps:next_id(maps(State2)), HeapSize),
                     {Ptr, State3} = write_heap_value(CalldataHeap, State2),
-                    aevm_eeevm_stack:push(Ptr, State3);
+                    CallDataGasCost = aevm_gas:mem_cost(byte_size(Calldata) div 32),
+                    State4 = set_gas(maps:get(gas, State3) - CallDataGasCost, State3),
```

Comment:
I would expect the gas accounting for the calldata to have already been paid by the caller (either contract call transaction, recursive call, contract create transaction - no contracts create another contract as of now).